eslint.config.js

```diff
@@ -0,0 +1,40 @@
+const tsPlugin = require("@typescript-eslint/eslint-plugin");
+const tsParser = require("@typescript-eslint/parser");
+const prettierRecommended = require("eslint-plugin-prettier/recommended");
+const globals = require("globals");
+const js = require("@eslint/js");
+
+module.exports = [
+    js.configs.recommended,
+    prettierRecommended,
+    {
+        ignores: ["node_modules", "dist"],
+    },
+    {
+        languageOptions: {
+            ecmaVersion: 2021,
+            globals: { ...globals.node, ...globals.browser, ...globals.es2015 },
```

Comment:
why do we use `es2015` for globals when the ecmaVersion is 2021?

Reply:
picking ecma 2021 as the baseline syntax

but using es6/es2015 as the minimum target for older browsers, so only es2015 globals will be available

Reply:
hmm on that note, module in tsconfig.json should be updated to match ecmaVersion